### PR TITLE
feat(command): ast-grep.executeAutofix

### PR DESCRIPTION
**Pull request by SoonIter**
the same feature as vscode-eslint
image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,9 +35,14 @@
...
   "contributes": {
     "commands": [
+      {
+        "command": "ast-grep.executeAutofix",
+        "title": "Fix all auto-fixable Problems",
+        "category": "ast-grep"
+      },
       {
         "command": "ast-grep.restartLanguageServer",
-        "title": "restart Language Server",
```
Comment (HerringtonDarkholme) [inline package.json:40]: nice typo fixing :)

File: src/extension/lsp.ts [hunk 2]
```diff
@@ -49,6 +50,27 @@ async function fileExists(pathFromRoot: string): Promise<boolean> {
...
+    version: textEditor.document.version,
+    text: textEditor.document.getText(),
+  }
+  const params = {
+    command: 'ast-grep.applyAllFixes',
+    arguments: [textDocument],
+  }
+  client.sendRequest(ExecuteCommandRequest.type, params).then(undefined, () => {
+    void window.showErrorMessage(
+      'Failed to apply Ast-grep fixes to the document. Please consider upgrading ast-grep version or opening an issue with steps to reproduce.',
```
Comment (HerringtonDarkholme) [inline src/extension/lsp.ts:69]: is void needed here? 
Also, showErrorMessage can have an action button to guide users installing the latest ast-grep version. (say by guiding users to the installing doc page.)

**Review by HerringtonDarkholme — APPROVED** [review]
Looks nice with a minor comment.

[LGTMeow](https://lgtmeow.com)

File: src/extension/lsp.ts (hunk 2 above)
Comment (SoonIter) [reply src/extension/lsp.ts]: done

**Pull request merged**